src/hardware/openbench-logic-sniffer/api.c

```diff
@@ -387,7 +387,16 @@ static int config_list(uint32_t key, GVariant **data,
 		return STD_CONFIG_LIST(key, data, sdi, cg, scanopts, drvopts,
 				       devopts);
 	case SR_CONF_SAMPLERATE:
-		*data = std_gvar_samplerates_steps(ARRAY_AND_SIZE(samplerates));
+		if (!sdi)
```

Comment:
Does this func ever get called with sdi==NULL ? If so, I'm worried about the previous block (SR_CONF_DEVICE_OPTIONS) where we use sdi without checking. Could be OK, I'm not sure

Reply:
Should be ok, fx2lafw returns SR_ERR_ARG if there's no sdi, too: https://sigrok.org/gitweb/?p=libsigrok.git;a=blob;f=src/hardware/fx2lafw/api.c;h=255788bfdd0731233ca75e0a9454bb7021ce5de2;hb=HEAD#l596

Other drivers don't even support listing the sample rate and just support get/set.